### PR TITLE
Updating methodology for how stripe customers are fetched

### PR DESCRIPTION
**Pull request by jackiemunroe**
This work refactors how we fetch the customers from stripe, adding validation and removing embedded stripe calls.

https://jira.mozilla.com/browse/SS-314

### CODE REVIEW

File: subhub/customer.py [hunk 1]
```diff
@@ -65,27 +65,38 @@ def create_customer(
...
+    return existing_payment_source(customer, source_token)
+
+
+def fetch_customer(subhub_account: SubHubAccount, user_id: str) -> stripe.Customer:
+    customer = None
+    db_account = subhub_account.get_user(user_id)
+    if db_account:
+        customer = stripe.Customer.retrieve(db_account.cust_id)
+        if "deleted" in customer and customer["deleted"]:
+            subhub_account.remove_from_db(user_id)
```
Comment (jackiemunroe) [inline subhub/customer.py]: This needs to be updated once Marty's "soft-delete" work has been completed and merged into master

Comment (marty331) [reply subhub/customer.py]: Soft delete is complete! @jackiemunroe

**Comment by bowlofstew** [comment]
Super clean code!  Minor conflict but LGTM other than that!

File: subhub/api/payments.py [hunk 2]
```diff
@@ -98,7 +97,7 @@ def check_stripe_subscriptions(customer: str) -> list:
         return []
 
 
-def sources_to_remove(subscriptions: list, customer: str) -> None:
+def sources_to_remove(subscriptions: list, customer: stripe.Customer) -> None:
```
Comment (sidler-mozilla) [inline subhub/api/payments.py]: consider`from stripe import Customer` to reduce all of the `stripe.Customer` -> `Customer`

File: subhub/tests/unit/test_payment_calls.py [hunk 3]
```diff
@@ -525,11 +527,40 @@ def test_update_payment_method_invalid_stripe_customer(
...
+                    }
+                ]
+            },
+        }
+    )
+
+    monkeypatch.setattr("flask.g.subhub_account", subhub_account)
+    monkeypatch.setattr("stripe.Customer.retrieve", stripe_customer)
+
+    (data, code) = payments.customer_update("user123")
```
Comment (sidler-mozilla) [inline subhub/tests/unit/test_payment_calls.py]: we just removed all of the `(x, y) = some_func()` expressions; let's be consistent; don't need the parens around the multiple items returned

File: subhub/customer.py [hunk 4]
```diff
@@ -65,27 +65,38 @@ def create_customer(
...
+            customer = None
+    return customer
 
 
-def existing_payment_source(customer_id: str, source_token: str) -> stripe.Customer:
-    existing_customer = stripe.Customer.retrieve(customer_id)
+def existing_payment_source(
+    existing_customer: stripe.Customer, source_token: str
+) -> stripe.Customer:
```
Comment (sidler-mozilla) [inline subhub/customer.py]: see changing `stripe.Customer` -> `Customer` might make this line NOT wrap

**Pull request merged**